src/Raven.Server/Json/WebsocketStream.cs

```diff
@@ -137,10 +147,18 @@ private void ThrowOnDisposed()
         public override void Flush()
         {
             ThrowOnDisposed();
+            if(_activeWriteTasks.Count > 0)
+                AsyncHelpers.RunSync(() =>
+                    Task.WhenAll(_activeWriteTasks)
+                        .ContinueWith(t => _activeWriteTasks.Clear(), _cancellationToken));
+        }
 
-            AsyncHelpers.RunSync(() =>
-                Task.WhenAll(_activeWriteTasks)
-                    .ContinueWith(t => _activeWriteTasks.Clear(), _cancellationToken));
+        public override async Task FlushAsync(CancellationToken cancellationToken)
+        {
+            if(_activeWriteTasks.Count > 0)
+                await Task.WhenAll(_activeWriteTasks)
```

Comment:
Why are you using ContinueWith? It is much harder to reason about. Just do mulitple awaits.